File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count=0;
        int ele=0;
        for(int i=0;i<nums.length;i++){
            if(count==0){
                ele=nums[i];
                count++;
            }
            else if(nums[i]==ele){
                count++;
            }
            else{
                count--;
            }
        }
        int cnt=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==ele){
                cnt++;
            }
        }
        if(cnt>nums.length/2){
            return ele;
        }
        return -1;

    }
}

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {

    let map= new Map();
    for(let i=0;i<nums.length;i++){
        let a=nums[i];
        if(map.has(a)){
            map.set(a,map.get(a)+1);
        }
        else{
            map.set(a,1);
        }
    }
    for(let key of map.keys()){
        let a=map.get(key);
        if(a>(nums.length/2)){
            return key;
        }
    }
    
};